src/collections/string.c

```diff
@@ -101,6 +101,13 @@ void _z_string_free(_z_string_t **str) {
     }
 }
 
+int _z_string_compare(const _z_string_t *left, const _z_string_t *right) {
+    if (_z_string_len(left) <= _z_string_len(right)) {
+        return strncmp(_z_string_data(left), _z_string_data(right), _z_string_len(left));
+    }
+    return strncmp(_z_string_data(left), _z_string_data(right), _z_string_len(right));
+}
+
```

Comment:
Use case for equal prefix with different length was missed.
For instance, "abc" and "abcd" would compare as equal.
This will be better:

```suggestion
int _z_string_compare(const _z_string_t *left, const _z_string_t *right) {
    size_t len_left = _z_string_len(left);
    size_t len_right = _z_string_len(right);

    int result = strncmp(_z_string_data(left), _z_string_data(right), len_left < len_right ? len_left : len_right);

    if (result == 0) {
        if (len_left < len_right) {
            return -1;
        } else if (len_left > len_right) {
            return 1;
        }
    }

    return result;
}

```